hivemind/averaging/allreduce.py

```diff
@@ -180,9 +182,10 @@ async def _generate_input_for_peer(self, peer_index: int) -> AsyncIterator[avera
             code=averaging_pb2.PART_FOR_AVERAGING,
             group_id=self.group_id,
             tensor_part=first_part,
+            metadata=self._weight_binary,
```

Comment:
Can we create a new `double weight;` field instead and avoid encoding the weight to the binary format manually?

Reply:
done so